src/olympia/editors/models.py

```diff
@@ -410,47 +413,86 @@ def get_key(cls, key=None, invalidate=False):
-            return getattr(amo, 'REVIEWED_ADDON_%s' % queue)
-        elif addon.type == amo.ADDON_DICT and queue:
-            return getattr(amo, 'REVIEWED_DICT_%s' % queue)
-        elif addon.type in [amo.ADDON_LPAPP, amo.ADDON_LPADDON] and queue:
-            return getattr(amo, 'REVIEWED_LP_%s' % queue)
-        elif addon.type == amo.ADDON_PERSONA:
-            return amo.REVIEWED_PERSONA
-        elif addon.type == amo.ADDON_SEARCH and queue:
-            return getattr(amo, 'REVIEWED_SEARCH_%s' % queue)
-        elif addon.type == amo.ADDON_THEME and queue:
-            return getattr(amo, 'REVIEWED_THEME_%s' % queue)
+        reviewed_score_name = None
+        if content_review:
+            # Content review always gives the same amount of points.
+            reviewed_score_name = 'REVIEWED_CONTENT_REVIEW'
+        elif post_review:
+            # There are 4 tiers of post-review scores depending on the addon
+            # weight.
+            try:
+                if version is None:
+                    raise AutoApprovalSummary.DoesNotExist
+                weight = version.autoapprovalsummary.weight
+            except AutoApprovalSummary.DoesNotExist as exp:
+                log.exception(
+                    'No such version/auto approval summary when determining '
+                    'event type to award points: %r', exp)
+                weight = 0
+            if weight >= amo.POST_REVIEW_WEIGHT_HIGHEST_RISK:
+                reviewed_score_name = 'REVIEWED_EXTENSION_HIGHEST_RISK'
+            elif weight >= amo.POST_REVIEW_WEIGHT_HIGH_RISK:
+                reviewed_score_name = 'REVIEWED_EXTENSION_HIGH_RISK'
+            elif weight >= amo.POST_REVIEW_WEIGHT_MEDIUM_RISK:
+                reviewed_score_name = 'REVIEWED_EXTENSION_MEDIUM_RISK'
+            else:
+                reviewed_score_name = 'REVIEWED_EXTENSION_LOW_RISK'
         else:
-            return None
+            if status == amo.STATUS_NOMINATED:
+                queue = 'FULL'
+            elif status == amo.STATUS_PUBLIC:
+                queue = 'UPDATE'
+            else:
+                queue = ''
+
+            if (addon.type in [amo.ADDON_EXTENSION, amo.ADDON_PLUGIN,
+                               amo.ADDON_API] and queue):
+                reviewed_score_name = 'REVIEWED_ADDON_%s' % queue
+            elif addon.type == amo.ADDON_DICT and queue:
+                reviewed_score_name = 'REVIEWED_DICT_%s' % queue
+            elif addon.type in [amo.ADDON_LPAPP, amo.ADDON_LPADDON] and queue:
+                reviewed_score_name = 'REVIEWED_LP_%s' % queue
+            elif addon.type == amo.ADDON_PERSONA:
+                reviewed_score_name = 'REVIEWED_PERSONA'
+            elif addon.type == amo.ADDON_SEARCH and queue:
+                reviewed_score_name = 'REVIEWED_SEARCH_%s' % queue
+            elif addon.type == amo.ADDON_THEME and queue:
+                reviewed_score_name = 'REVIEWED_THEME_%s' % queue
+
```

Comment:
You could use format strings like `'REVIEWED_THEME_{queue}` etc and use `.format(queue=queue)` which ignores missing format strings so that `REVIEWED_PERSONA` won't raise an exception.

Reply:
That does not buy us much though, because if `queue` is not present for all non-persona conditions, we don't want to award points. Ultimately I tried not to refactor this too much, I'm likely to get it wrong :)